pkg/jsonnetfile/jsonnetfile.go

```diff
@@ -19,44 +19,84 @@ import (
 	"io/ioutil"
 	"os"
 
+	v0 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v0"
+	v1 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v1"
+	depsv1 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v1/deps"
 	"github.com/pkg/errors"
-
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec"
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec/deps"
 )
 
 const (
 	File     = "jsonnetfile.json"
 	LockFile = "jsonnetfile.lock.json"
 )
 
-var ErrNoFile = errors.New("no jsonnetfile")
+var (
+	ErrNoFile   = errors.New("no jsonnetfile")
+	ErrUpdateJB = errors.New("jsonnetfile version unknown, update jb")
+)
 
 // Load reads a jsonnetfile.(lock).json from disk
-func Load(filepath string) (spec.JsonnetFile, error) {
+func Load(filepath string) (v1.JsonnetFile, error) {
 	bytes, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		return spec.New(), err
+		return v1.New(), err
 	}
 
 	return Unmarshal(bytes)
 }
 
 // Unmarshal creates a spec.JsonnetFile from bytes. Empty bytes
 // will create an empty spec.
-func Unmarshal(bytes []byte) (spec.JsonnetFile, error) {
-	m := spec.New()
+func Unmarshal(bytes []byte) (v1.JsonnetFile, error) {
+	m := v1.New()
+
 	if len(bytes) == 0 {
 		return m, nil
 	}
-	if err := json.Unmarshal(bytes, &m); err != nil {
-		return m, errors.Wrap(err, "failed to unmarshal file")
+
+	versions := struct {
+		Version float64 `json:"version"`
+	}{}
+
+	err := json.Unmarshal(bytes, &versions)
+	if err != nil {
+		return m, err
 	}
-	if m.Dependencies == nil {
-		m.Dependencies = make(map[string]deps.Dependency)
+
+	if versions.Version > 1 {
```

Comment:
I think a `switch` would really improve readability here:

```go
switch v := versions.Version {
  case 0:
    // do legacy unmarshalling
  case 1:
    json.Unmarshal(bytes, &m)
  default:
    return nil, fmt.Errorf("Unknown version %v. Try updating jb.", v)
}
```

Also the version numbers should be constants:
```go
package v1
const Number = 1
```

Reply:
We don't have the 0 version as that simply doesn't have a version at all. I think the current way is the simplest and most straightforward way to find the correct version to unmarshal.

Reply:
I'm afraid I don't agree.

Although "version 0" isn't explicitely defined, it marks all jsonnetfile schemas before we started versioning them. And as they are all compatible (there were no backwards-incompatible changes between 0.1 and 0.2), I think it's fine to call this "version 0". Also, the package for parsing that is called `v0` as well.

That being said, I think the big value of a switch is that you easily realize what possible cases your logic has, instead of having to read two if statements, where it might be easy to miss what actually happens (realized what actually happens on the 3rd time reading it on an iPad, so had quite a hard time).

Also, if we added another version (`2`), that `if else` chain would become even more complicated .... which is exactly why Go has general (empty) switches.